File: app/src/main/java/com/kieran/winnipegbusbackend/RouteSchedule.java
package com.kieran.winnipegbusbackend;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class RouteSchedule extends Route implements Serializable {
    private final static String SCHEDULED_STOPS_TAG = "scheduled-stops";

    private List<ScheduledStop> stops;

    public RouteSchedule(RouteSchedule routeSchedule) {
        super(routeSchedule);
    }

    public RouteSchedule(JSONObject jsonObject) {
        super(jsonObject);

        stops = new ArrayList<>();

        loadScheduledStops(jsonObject);
    }

    private void loadScheduledStops(JSONObject jsonObject) {
        try {
            JSONArray scheduledStops = jsonObject.getJSONArray(SCHEDULED_STOPS_TAG);

            for (int s = 0; s < scheduledStops.length(); s++) {
                JSONObject stop = scheduledStops.getJSONObject(s);
                try {
                    stops.add(new ScheduledStop(stop, this));
                }catch (Exception e) {
                    //blank
                }
            }

        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public List<ScheduledStop> getScheduledStops() {
        return stops;
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/RouteSchedule.kt
package com.kieran.winnipegbusbackend

import org.json.JSONException
import org.json.JSONObject
import java.io.Serializable
import java.util.*

class RouteSchedule : Route, Serializable {
    fun compareTo(other: Nothing): Int {
        return super.compareTo(other)
    }

    private var stops: ArrayList<ScheduledStop>? = null

    val scheduledStops: List<ScheduledStop>?
        get() = stops

    constructor(routeSchedule: RouteSchedule) : super(routeSchedule) {}

    constructor(jsonObject: JSONObject) : super(jsonObject) {

        stops = ArrayList()

        loadScheduledStops(jsonObject)
    }

    private fun loadScheduledStops(jsonObject: JSONObject) {
        try {
            val scheduledStops = jsonObject.getJSONArray(SCHEDULED_STOPS_TAG)

            (0 until scheduledStops.length())
                    .map { scheduledStops.getJSONObject(it) }
                    .forEach {
                        try {
                            stops?.add(ScheduledStop(it, this))
                        } catch (e: Exception) {
                            //blank
                        }
                    }

        } catch (e: JSONException) {
            e.printStackTrace()
        }

    }

    companion object {
        private val SCHEDULED_STOPS_TAG = "scheduled-stops"
    }
}